core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala

```diff
@@ -93,7 +93,8 @@ private[history] class HistoryPage(parent: HistoryServer) extends WebUIPage("")
             } else if (requestedIncomplete) {
               <h4>No incomplete applications found!</h4>
             } else {
-              <h4>No completed applications found!</h4> ++
+              <h4>No {if (requestedIncomplete) "running" else "completed"}
```

Comment:
You can revert all changes to this file. This issue has been fixed separately (as you can see in the condition in L93).